### PR TITLE
Remove `ComponentWrapper` and add `EditorMaskWrapper` in `Editor`.

### PR DESCRIPTION
**Pull request by tanbowensg**
截屏2022-02-10 下午2 51 36

# Todo
1. add refs to arco components
2. does not support components in dialog now

### CODE REVIEW

File: packages/runtime/src/App.tsx [hunk 1]
```diff
@@ -1,34 +1,46 @@
...
 import { ImplWrapper } from './components/_internal/ImplWrapper';
-import { AppProps, UIServices } from './types';
+import { AppProps, UIServices, AppLifeCycles } from './types';
 import { DebugEvent, DebugStore } from './services/DebugComponents';
 import { RuntimeAppSchemaManager } from './services/RuntimeAppSchemaManager';
 import { resolveChildrenMap } from './utils/resolveChildrenMap';
 
 // inject modules to App
-export function genApp(services: UIServices) {
+export function genApp(services: UIServices, lifeCycles?: AppLifeCycles) {
```
Comment (Yuyz0112) [inline packages/runtime/src/App.tsx:10]: Does `hooks` sounds like a more general naming of this feature?

**Review by Yuyz0112 — CHANGES_REQUESTED** [review]
Good job!

Two questions:
1. Shall we add some unit tests to the math part?
2. What's the behavior if a component does not implement $ref?

File: packages/chakra-ui-lib/src/components/Box.tsx [hunk 2]
```diff
@@ -295,7 +295,7 @@ export default implementRuntimeComponent({
     styleSlots: ['content'],
     events: [],
   },
-})(({ customStyle, slotsElements, ...restProps }) => {
+})(({ customStyle, slotsElements, $ref, ...restProps }) => {
```
[the remark this answers is not in the document]
Comment (tanbowensg) [reply packages/chakra-ui-lib/src/components/Box.tsx:298]: Just want to distinguish with `ref`. Maybe change a name, like `elementRef`?

File: packages/chakra-ui-lib/src/components/Button.tsx [hunk 3]
```diff
@@ -55,6 +55,7 @@ export default implementRuntimeComponent({
     colorScheme,
     isLoading,
     customStyle,
+    $onRef,
```
[the remark this answers is not in the document]
Comment (tanbowensg) [reply packages/chakra-ui-lib/src/components/Button.tsx:58]: How about `getElement`?

**Comment by tanbowensg** [comment]
> Good job!
> 
> Two questions:
> 
> 1. Shall we add some unit tests to the math part?
> 2. What's the behavior if a component does not implement $ref?

1. ok.
2. It can not be hoverd, selected or drop components.

**Review by Yuyz0112 — APPROVED** [review]

**Pull request merged**